app/presenters/agent_presenter.rb

```diff
@@ -10,6 +10,10 @@ def self.terms
     ] + CitiResourceTerms.all
   end
 
+  def display_label
```

Comment:
Can you try setting `self.human_readable_type = 'Artist'` under `app/models/agent.rb` that should accomplish the same thing and make it more universal.